test/unit-test/rtp_api/rtp_api_utest.c

```diff
@@ -0,0 +1,892 @@
+/* Unity includes. */
+#include "unity.h"
+#include "catch_assert.h"
+
+/* Standard includes. */
+#include <string.h>
+#include <stdint.h>
+
+/* API includes. */
+#include "rtp_api.h"
+#include "rtp_data_types.h"
+#include "rtp_endianness.h"
+
+
+/* ===========================  EXTERN VARIABLES  =========================== */
+
+#define MAX_FRAME_LENGTH        10 * 1024
+#define RTP_HEADER_MIN_LENGTH   12 /* No CSRC and no extension. */
+
+uint8_t buffer[ MAX_FRAME_LENGTH ];
+
+void setUp( void )
+{
+    memset( &( buffer[ 0 ] ),
+            0,
+            sizeof( buffer ) );
+}
+
+void tearDown( void )
+{
+}
+
+/* ==============================  Test Cases  ============================== */
+
+/**
+ * @brief Validate Rtp_Init in case of valid inputs.
+ */
+void test_Rtp_Init_Pass( void )
+{
+    RtpResult_t result;
+    RtpContext_t ctx = { 0 };
+
+    result = Rtp_Init( &ctx );
+
+    TEST_ASSERT_EQUAL( RTP_RESULT_OK,
+                       result );
+
+    TEST_ASSERT_NOT_NULL( ctx.readWriteFunctions.readUint32Fn );
+
+    TEST_ASSERT_NOT_NULL( ctx.readWriteFunctions.writeUint32Fn );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate RTP_Serialize in case of valid inputs.
+ */
+void test_Rtp_Serialize_Pass( void )
+{
+    RtpResult_t result;
+    RtpContext_t ctx = { 0 };
+    RtpPacket_t packet = { 0 };
+    uint8_t buffer[ MAX_FRAME_LENGTH ] = { 0 };
+    size_t length = MAX_FRAME_LENGTH;
+
+    result = Rtp_Init( &ctx );
+
+    TEST_ASSERT_EQUAL( RTP_RESULT_OK,
+                       result );
+
+    packet.header.payloadType = 96;
+    packet.header.sequenceNumber = 1234;
+    packet.header.timestamp = 0x12345678;
+    packet.header.ssrc = 0x87654321;
+    packet.payloadLength = 11;
+    packet.pPayload = ( uint8_t * ) "hello world";
+
+    result = Rtp_Serialize( &ctx,
+                            &packet,
+                            buffer,
+                            &length );
+
+    TEST_ASSERT_EQUAL( RTP_RESULT_OK,
+                       result );
+
+    TEST_ASSERT_EQUAL( 23,
+                       length );
+
+    TEST_ASSERT_EQUAL_UINT8_ARRAY( ( uint8_t * ) "\x80\x60\x04\xD2\x12\x34\x56\x78\x87\x65\x43\x21\x68\x65\x6C\x6C\x6F\x20\x77\x6F\x72\x6C\x64",
+                                   buffer,
+                                   length );
```

Comment:
Can you help move this expected result into a variable with comments? That would be easier to read.
For example:
```
uint8_t expectedSerializedBuffer = {
    0x80, ... /* RTP header */
    ... /* timestamp/ssrc/etc */
    ... /* RTP payload - "hello world" */
}

...

TEST_ASSERT_EQUAL_UINT8_ARRAY( expectedSerializedBuffer,
                                   buffer,
                                   length );
```